addons/website_slides_survey/models/slide_channel.py

```diff
@@ -1,9 +1,38 @@
 # -*- coding: utf-8 -*-
 # Part of Odoo. See LICENSE file for full copyright and licensing details.
 
-from odoo import fields, models
+from odoo import api, fields, models, _
 from odoo.osv import expression
 
+class ChannelUsersRelation(models.Model):
+    _inherit = 'slide.channel.partner'
+
+    certification_passed = fields.Boolean('Certified', compute='_compute_certification_passed', search='_search_certification_passed')
+    nbr_certification = fields.Integer(related='channel_id.nbr_certification')
+
+    @api.depends('channel_id.slide_partner_ids.survey_scoring_success')
+    def _compute_certification_passed(self):
+        base_domain = expression.OR([
+            [('channel_id', '=', slide_channel.channel_id.id),
+             ('partner_id', '=', slide_channel.partner_id.id)]
+            for slide_channel in self
+        ])
+        certified = self.env['slide.slide.partner'].sudo()._read_group(
+            domain=expression.AND([base_domain, [('survey_scoring_success', '=', True)]]),
+            groupby=['channel_id'],
+            aggregates=['partner_id:recordset']
+        )
+        mapped_data = {channel_id: partner_ids for channel_id, partner_ids in certified}
+        for member in self:
+            member.certification_passed = member.partner_id in mapped_data.get(member.channel_id, [])
+
+    @api.model
+    def _search_certification_passed(self, operator, value):
+        domain = [('survey_scoring_success', '=', True)]
+        if operator == 'in' and isinstance(value, list) and value:
+            domain = expression.AND([domain, [('channel_id', 'in', value)]])
+        certified_partners = self.env['slide.slide.partner'].sudo().search(domain)
```

Comment:
I would try making a Query object, starting from a search on `self` and adding a `JOIN`  (see `add_join`) on `slide.slide.partner` for the condition(s) we need; then you can return `[('id', 'in', query)]`. 

This will also make sure we do not return memberships for courses X and Y for students A and B if both are registered but A only passed X and B only passed Y, which looks like it could be the case here as `certified_partners.partner_id.ids` will return `[A.id, B.id]` and `certified_partners.channel_id.ids` `[X.id, Y.id]`.
In any case, let's have a test for this :)